lib/matplotlib/dates.py

```diff
@@ -1254,6 +1241,8 @@ def __init__(self, bymonthday=None, interval=1, tz=None):
 
         Default is to tick every day of the month: ``bymonthday=range(1,32)``
         """
+        if not isinstance(interval, int) or interval < 1:
```

Comment:
It has a chance of breaking user code.  This also breaks in the case of non-python ints, ex

```python
In [56]: d = np.uint8(5)

In [57]: isinstance(d, int)
Out[59]: False

In [60]: d == int(d)
Out[60]: True
```

`isinstance` should only be used when there is no other reasonable option.